rootfs/api/models/build.py

```diff
@@ -78,9 +79,37 @@ def create(self, user, *args, **kwargs):
             raise DeisException(str(e)) from e
 
     def save(self, **kwargs):
-        removed = {}
         previous_release = self.app.release_set.filter(failed=False).latest()
-        if previous_release.build is not None:
+
+        if (
+            settings.DEIS_DEPLOY_REJECT_IF_PROCFILE_MISSING is True and
+            # previous release had a Procfile and the current one does not
+            (
+                previous_release.build is not None and
+                len(previous_release.build.procfile) > 0 and
+                len(self.procfile) == 0
+            )
+        ):
+            # Reject deployment
+            raise Conflict(
+                'Last deployment had a Procfile but is missing in this deploy. '
+                'For a successful deployment provide a Procfile.'
+            )
+
+        # See if processes are permitted to be removed
+        remove_procs = (
+            # If set to True then contents of Procfile does not affect the outcome
+            settings.DEIS_DEPLOY_PROCFILE_MISSING_REMOVE is True or
```

Comment:
@kmala those 2 settings do 2 different things. One is all about rejecting if you are missing a `Procfile` and the other is about letting a deploy through without scaling down any procs types if you miss the `Procfile`

I'm not sure I am following your reasoning around both of those settings being set to false - Is that something you tried doing?

I could do the reject only option or we could copy the procfile info between builds if the fear is the Build object now has empty `Procfile` field (a valid concern to have).